assets/javascripts/discourse/components/bar-chart.js.es6

```diff
@@ -0,0 +1,96 @@
+import loadScript from "discourse/lib/load-script";
+import { default as computed } from "discourse-common/utils/decorators";
+import themeColor from "../lib/themeColor";
+
+export default Ember.Component.extend({
+  barsColor: themeColor("--tertiary"),
+  barsHoverColor: themeColor("--tertiary-high"),
+  gridColor: themeColor("--primary-low"),
+  labelsColor: themeColor("--primary-medium"),
+  chart: undefined,
+
+  @computed("data", "options")
+  config(data, options) {
+    return {
+      type: "bar",
+      data: data,
+      options: options,
+    };
+  },
+
+  @computed("labels.[]", "values.[]", "datasetName")
+  data(labels, values, datasetName) {
+    return {
+      labels: labels,
+      datasets: [
+        {
+          label: datasetName,
+          data: values,
+          backgroundColor: this.barsColor,
+          hoverBackgroundColor: this.barsHoverColor,
+        },
+      ],
+    };
+  },
+
+  @computed
+  options() {
+    return {
+      scales: {
+        legend: {
+          labels: {
+            fontColor: this.labelsColor,
+          },
+        },
+        xAxes: [
+          {
+            gridLines: {
+              color: this.gridColor,
+              zeroLineColor: this.gridColor,
+            },
+            ticks: {
+              fontColor: this.labelsColor,
+            },
+          },
+        ],
+        yAxes: [
+          {
+            gridLines: {
+              color: this.gridColor,
+              zeroLineColor: this.gridColor,
+            },
+            ticks: {
+              beginAtZero: true,
+              fontColor: this.labelsColor,
+            },
+          },
+        ],
+      },
+    };
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+    this._initChart();
+  },
+
+  didUpdate() {
+    this._super(...arguments);
+    this.chart.data = this.get("data");
+    this.chart.update();
+  },
```

Comment:
Right now, this hook never calls because the query-result component always reinserts after querying data. But it's better to have it here to escape regressions in the future.

Reply:
I'd rather have a test than some code that is unused but might be used. Much better at preventing regressions as well 😉

Reply:
I agree that it's better to have a test. We could check that the chart is updating when we're performing the query one more time. Unfortunately, the chart is rendered on a canvas, and it's hard to check if it's updated. It's possible to [check pixels on a canvas](https://www.npmjs.com/package/qunit-assert-canvas), but it looks overcomplicated to me.

Probably we can remove this hook. But then the problem would appear if we stop reinserting the whole component with query results after every query or if someone uses this code or the component itself for a chart that should be updated without reinserting.